enhancements/microshift/microshift-pluggable-core-components.md

```diff
@@ -0,0 +1,381 @@
+  could implement its own rebase script.
+
+* Incompatible component rpms should be impossible to install together.
+  i.e. `microshift-networking-bridge` plus `microshift-networking-ovnk`,
+  this can be handled with Conflict spec clauses. (This is just an example,
+  since microshift-networking-bridge would not be necessary as it's the
+  default for container-networking)
+
+* Systemd services of the then optional networking components (like `microshift-ovs-init`)
+  should still start before the `microshift` systemd service, but `microshift` cannot
+  depend on `microshift-ovs-init` anymore. This can be acomplished by using the
+  Before=microshift clause in `microshift-ovs-init.service` and enabling this service
+  during the `microshift-networking-ovnk` sub-package install.
+  An alternative to this can be making `microshift-ovs-init` part of the plugin.
+
+#### Plugin calling convention
+
+The `plugin` binaries receive configuration details via environment variables, using the same variable format
+MicroShift already receives (see MicroShift's howto_config.md):
+
+| Environment Variable                    | Description                                                       |
+|-----------------------------------------|-------------------------------------------------------------------|
+| MICROSHIFT_CLUSTER_CLUSTERCIDR          | A block of IP addresses from which Pod IP addresses are allocated |
+| MICROSHIFT_CLUSTER_SERVICECIDR          | A block of virtual IP addresses for Kubernetes services           |
+| MICROSHIFT_CLUSTER_SERVICENODEPORTRANGE | The port range allowed for Kubernetes services of type NodePort   |
+| MICROSHIFT_CLUSTER_URL                  | URL of the API server for the cluster.                            |
+| MICROSHIFT_NODEIP                       | The IP address of the node, defaults to IP of the default route   |
+| MICROSHIFT_SUBJECTALTNAMES              | Subject Alternative Names for apiserver certificates              |
+| MICROSHIFT_LOGVLEVEL                    | Log verbosity (Normal, Debug, Trace, TraceAll)                    |
+| MICROSHIFT_NODENAME                     | The name of the node, defaults to hostname                        |
+| MICROSHIFT_BASEDOMAIN                   | Base DNS domain used to construct fully qualified router and API domain names. |
+
+Plugins can read their own configuration files in case of needing more specific
+configuration details, or this interface could be expanded in the future to
+provide more generic data which plugins may need.
+
+As input `plugin` will receive an argument command.
+
+| Command                    | Description                                                       |
+|-----------------------------------------|-------------------------------------------------------------------|
+| api-version | Identifies the version of the calling convention for this plugin, this could be useful if we the implementation needed future changes |
```

Comment:
This sentence seems incomplete.

Reply:
the "we" needs to be removed.
```suggestion
| api-version | Identifies the version of the calling convention for this plugin, this could be useful if the implementation needed future changes |
```